library/ssl_tls12_server.c

```diff
@@ -2170,11 +2170,65 @@ static void ssl_handle_id_based_session_resumption(mbedtls_ssl_context *ssl)
 }
 
 MBEDTLS_CHECK_RETURN_CRITICAL
-static int ssl_write_server_hello(mbedtls_ssl_context *ssl)
+static int ssl_generate_random(mbedtls_ssl_context *ssl)
```

Comment:
This first commit does to many things at once, in particular moving code and changing code. Please rework it with commits doing one thing at a time and with commit messages covering all the aspects of the changes in each commit.

Reply:
Will re-write the history.

Reply:
The history has been re-written